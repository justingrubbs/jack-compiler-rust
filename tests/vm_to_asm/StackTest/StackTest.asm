@256
D=A
@SP
M=D
@StackTest.Sys.init$ret.1
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Sys.init
0;JMP
(StackTest.Sys.init$ret.1)
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@EQ_true_2
D;JEQ
@SP
A=M-1
M=0
@EQ_end_2
0;JMP
(EQ_true_2)
@SP
A=M-1
M=-1
(EQ_end_2)
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@EQ_true_3
D;JEQ
@SP
A=M-1
M=0
@EQ_end_3
0;JMP
(EQ_true_3)
@SP
A=M-1
M=-1
(EQ_end_3)
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@EQ_true_4
D;JEQ
@SP
A=M-1
M=0
@EQ_end_4
0;JMP
(EQ_true_4)
@SP
A=M-1
M=-1
(EQ_end_4)
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@LT_true_5
D;JLT
@SP
A=M-1
M=0
@LT_end_5
0;JMP
(LT_true_5)
@SP
A=M-1
M=-1
(LT_end_5)
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@LT_true_6
D;JLT
@SP
A=M-1
M=0
@LT_end_6
0;JMP
(LT_true_6)
@SP
A=M-1
M=-1
(LT_end_6)
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@LT_true_7
D;JLT
@SP
A=M-1
M=0
@LT_end_7
0;JMP
(LT_true_7)
@SP
A=M-1
M=-1
(LT_end_7)
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@GT_true_8
D;JGT
@SP
A=M-1
M=0
@GT_end_8
0;JMP
(GT_true_8)
@SP
A=M-1
M=-1
(GT_end_8)
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@GT_true_9
D;JGT
@SP
A=M-1
M=0
@GT_end_9
0;JMP
(GT_true_9)
@SP
A=M-1
M=-1
(GT_end_9)
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
D=M-D
@GT_true_10
D;JGT
@SP
A=M-1
M=0
@GT_end_10
0;JMP
(GT_true_10)
@SP
A=M-1
M=-1
(GT_end_10)
@57
D=A
@SP
A=M
M=D
@SP
M=M+1
@31
D=A
@SP
A=M
M=D
@SP
M=M+1
@53
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
M=D+M
@112
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
M=M-D
@SP
A=M-1
M=-M
@SP
AM=M-1
D=M
A=A-1
M=D&M
@82
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
M=D|M
@SP
A=M-1
M=!M
