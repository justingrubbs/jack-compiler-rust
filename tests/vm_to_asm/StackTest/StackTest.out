|  RAM[0]  | RAM[256] | RAM[257] | RAM[258] | RAM[259] | RAM[260] |
|     370  |      51  |     261  |     256  |    3030  |    3004  |
